packages/jest-cli/src/watch.js

```diff
@@ -36,19 +35,17 @@ let hasExitListener = false;
 const watch = (
   initialGlobalConfig: GlobalConfig,
   contexts: Array<Context>,
-  argv: Argv,
-  pipe: stream$Writable | tty$WriteStream,
+  outputStream: stream$Writable | tty$WriteStream,
   hasteMapInstances: Array<HasteMap>,
   stdin?: stream$Readable | tty$ReadStream = process.stdin,
 ) => {
-  updateArgv(argv, argv.watch ? 'watch' : 'watchAll', {
-    testNamePattern: argv.testNamePattern,
-    testPathPattern: argv.testPathPattern || (argv._ || []).join('|'),
-  });
```

Comment:
Because of lack of this update logic, `globalConfig.onlyChanged` is not updated properly on the first run, resulting in tests being run, despite there's nothing changed in git.